src/client.ts

```diff
@@ -81,21 +84,40 @@ export class PlainClient {
     return unwrapData(res, (q) => q.customer);
   }
 
+  /**
+   * If the customer is not found this will return null.
+   */
+  async getCustomerByEmail(
+    variables: VariablesOf<typeof CustomerByEmailDocument>
+  ): SDKResult<CustomerPartsFragment | null> {
+    const res = await request(this.#ctx, {
+      query: CustomerByEmailDocument,
+      variables,
+    });
+
+    return unwrapData(res, (q) => q.customerByEmail);
+  }
+
   /**
    * Allows you to create or update a customer. If you need to get the customer id
    * for a customer in Plain, this is typically your first step.
    */
   async upsertCustomer(
     input: VariablesOf<typeof UpsertCustomerDocument>['input']
-  ): SDKResult<CustomerPartsFragment> {
+  ): SDKResult<{ result: UpsertResult; customer: CustomerPartsFragment }> {
```

Comment:
maybe 
```suggestion
  ): SDKResult<{ upsertResult: UpsertResult; customer: CustomerPartsFragment }> {
```

would be more verbose 🤷

Reply:
I agree but I don't want to deviate from our graphql naming!